sklearn/utils/testing.py

```diff
@@ -372,14 +375,30 @@ def __exit__(self, *exc_info):
     assert_greater = _assert_greater
 
 
+def _assert_sparse_allclose(actual, desired, rtol=1e-7, atol=0, err_msg='',
+                            verbose=True):
+    """Check if two sparse arrays are equal up to the desired tolerance.
+
+    It compares the difference between `actual` and `desired` to
```

Comment:
Do you mean absolute difference as in the absolute difference between -1 and 1 is 0?
(I am pretty sure I misunderstood what u said... could you please expand a bit?)